jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/Fabric8HttpUtil.java

```diff
@@ -71,4 +71,23 @@ public static String toFormData(Map<String, String> formData) throws Unsupported
     }
     return result.toString();
   }
+
+  /**
+   * Parse www-authentication scope and append additional scope value if not present. It follows
+   * format specified in <a href="https://docs.docker.com/registry/spec/auth/token/#how-to-authenticate">Docker documentation.</a>
+   *
+   * @param scopeStr original scope string from www-authenticate map
+   * @param value value to append in scope string
+   * @return updated scope string
+   */
+  public static String appendAdditionalScopeIfNotPresent(String scopeStr, String value) {
+    String[] scopeParts = scopeStr.split(":");
+    if (scopeParts.length > 2) {
+      String pullOrPushScope = scopeParts[2];
+      if (!pullOrPushScope.contains(value)) {
+        scopeParts[2] = scopeParts[2] + "," + value;
+      }
+    }
+    return String.join(":", scopeParts);
+  }
```

Comment:
This is still not OK.

See the ref for [WWW-Authenticate](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate)

We first need to identify the appropriate `WWW-Authenticate` header and challenge for the current realm.

Then it might be interesting to parse the challenges as a Map or more specifically as Properties and then retrieve the one that we're interested in (`scope`).

Reply:
I wouldn't say that that method is parsing the header and reading properties

Reply:
OK, I think I'm getting confused here.
In any case, that method might benefit from some built-in Java property parsing.
In addition, the signature for the return should probably be changed to something like:
```java
public static List<Properties> extractAuthenticationChallengeIntoMap(HttpResponse<?> response)
```

Reply:
Please also check the [WWW-Authenticate](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate) spec documentation.

Probably that's why the parsing is not being done correctly and that might be clearly the source of the problem.